### ISSUE TITLE
Support authenticating with a gnosis safe

### ISSUE THREAD

**Issue by topocount** [coordinape/coordinape#1253]
@reeserj Gnosis doesnt support EIP 1271 signatures. We can work around this by using the web-react gnosis-safe connector, but we don't use a version that contains this connector yet.

_Originally posted by @topocount in https://github.com/coordinape/coordinape/issues/1242#issuecomment-1209675844_

**Comment by spydermonkee** [comment]
@karelianpie - is this tricky one of any interest to you?

**Comment by karelianpie** [comment]
I'd be happy to, @topocount do you have some time to give me some context on this?

**Comment by topocount** [comment]
@karelianpie This basically involves porting our app to work as a gnosis safe app. The biggest details, are 

1. setting up a gnosis safe connector in our web3-react setup here: https://github.com/coordinape/coordinape/blob/58702f9ec1463ed2ac2ea73db5d7f58fdb0864f4/src/components/WalletAuthModal/WalletAuthModal.tsx#L168-L200
2. Creating a manifest.json file as documented here: https://docs.gnosis-safe.io/build/sdks/safe-apps/releasing-your-safe-app#get-your-safe-app-listed-in-the-safe

There might be some tweaking beyond that to get this working within the gnosis safe app context, but I'm not sure.

Let me know if you still want to chat.

**Comment by karelianpie** [comment]
Started integrating this, currently trying to test locally but having an issue with the `manifest.json` structure, contacted their support and was instructed to leave a question on StackExchange https://ethereum.stackexchange.com/questions/133641/app-manifest-does-not-fulfil-the-required-structure

Also trying to follow up on their discord server https://discord.com/channels/907968493134155807/908129365345312768/1008845713079271464

**Comment by levity** [comment]
@topocount, i was just doing some catchup & review of what you and @karelianpie have looked at so far. if i'm understanding things correctly, it seems like we're really just trying to create a Safe App, and Zodiac isn't exactly the right fit for us? it seems like Zodiac is more focused on low-level smart contract functionality, and you still need something to provide a UI, as in the case of [Zodiac Exit](https://gnosis.github.io/zodiac/docs/tutorial-module-exit-app/get-started).

**Comment by levity** [comment]
But then, a Safe App is just a web3 app that's been greenlisted to show up in the Safe UI, so perhaps the more significant work here would be to change our app to reduce the number of transactions necessary, where possible? For a multisig, it would still require signers to coordinate in real time to use the app.

**Comment by karelianpie** [comment]
@levity for this one I hit a road block/knowledge gap when it came to coordinape's contracts as I'm still learning and not very experienced yet in Solidity and was waiting for someone from Core to kind of help me from there. I was a bit out of my depth when it came to knowledge of current coordinape's SC and solidity. I took this discovery in order to further my knowledge of Zodiac, Gnosis Safe and Solidity/SM but as I said, unable at this point to come up with a project plan.

Would like to stay in the loop if someone takes this further as now I have a little of background context and could try and help out where possible, however, not experienced enough with SC to add much value at the minute .

**Comment by levity** [comment]
> @levity for this one I hit a road block/knowledge gap when it came to coordinape's contracts as I'm still learning and not very experienced yet in Solidity and was waiting for someone from Core to kind of help me from there. I was a bit out of my depth when it came to knowledge of current coordinape's SC and solidity. I took this discovery in order to further my knowledge of Zodiac, Gnosis Safe and Solidity/SM but as I said, unable at this point to come up with a project plan.
> 
> Would like to stay in the loop if someone takes this further as now I have a little of background context and could try and help out where possible, however, not experienced enough with SC to add much value at the minute .

yep, makes sense, no worries, thanks for the assistance up to this point! will keep you in the loop
